### DISCUSSION TITLE
how to show the smae page for two route in nextjs13 app directory ?

### DISCUSSION THREAD

**Discussion by zwelhtetyan** [vercel/next.js#48610] [Help]
### Summary

Firstly ,  I want to show all menus on home page (http://localhost:3000)

![Screenshot from 2023-04-20 10-29-01](https://user-images.githubusercontent.com/89287841/233254732-52530895-cf88-407d-bfe0-9256157b959c.png)

But I want to show the same page also for (https://localhost:3000/menus)

how can I achieve this without reusing the same component.

**Comment by haydenbleasel** [comment]
@zwelhtetyan There are two solutions here I can think of:

1. Move all the page logic into a component e.g. `<Menus />, then on each page, simply render `<Menus />` for the same content.
2. Implement a [rewrite](https://nextjs.org/docs/api-reference/next.config.js/rewrites) in your `next.config.js`.

In your case, sounds like the latter is preferable.

> Rewrites act as a URL proxy and mask the destination path, making it appear the user hasn't changed their location on the site. In contrast, [redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects) will reroute to a new page and show the URL changes.
